It seems Studio and GCC have different opinions about xpcom symbols 
visibility. Unhide xpcom_glue. Complements xpcom-glue-no-hidden.patch

--- mozilla-esr24/xpcom/base/nscore.h.~1~	2014-07-08 09:58:13.430177399 +0400
+++ mozilla-esr24/xpcom/base/nscore.h	2014-07-08 09:58:50.883016452 +0400
@@ -261,11 +261,7 @@
 #  define nsAString nsAString_internal
 #  define nsACString nsACString_internal
 #else
-#  ifdef HAVE_VISIBILITY_ATTRIBUTE
-#    define NS_COM_GLUE NS_VISIBILITY_HIDDEN
-#  else
-#    define NS_COM_GLUE
-#  endif
+#  define NS_COM_GLUE
 #endif
 
 #if (defined(DEBUG) || defined(FORCE_BUILD_REFCNT_LOGGING))
